src/object/object_primitive.c

```diff
@@ -17456,12 +17459,29 @@ mr_data_writeval_json (OR_BUF * buf, DB_VALUE * value)
   int rc = NO_ERROR;
   DB_VALUE json_body, schema_raw;
 
+  db_make_null (&json_body);
+  db_make_null (&schema_raw);
+
   if (value->data.json.json_body == NULL || DB_IS_NULL (value))
     {
       assert (false);
       return ER_FAILED;
     }
 
+  if (buf->error_abort)
+    {
+      int estimated_length = mr_data_lengthval_json (value, true);
+
+      if (estimated_length > buf->endptr - buf->ptr)
```

Comment:
it is better to cast both LHS and RHS as `ptrdiff_t`.
please refer to `mr_data_writeval_set` for an example.

Reply:
done :)